src/components/ha-textfield.ts

```diff
@@ -54,11 +54,19 @@ export class HaTextField extends TextFieldBase {
       .mdc-text-field:not(.mdc-text-field--with-leading-icon) {
         padding: var(--text-field-padding, 0px 16px);
       }
+      .mdc-text-field--with-leading-icon {
+        padding-inline-start: initial;
+        padding-inline-end: 16px;
+        direction: var(--direction);
+      }
       .mdc-text-field__affix--suffix {
         padding-left: var(--text-field-suffix-padding-left, 12px);
         padding-right: var(--text-field-suffix-padding-right, 0px);
-        padding-inline-start: var(--text-field-suffix-padding-left, 12px);
-        padding-inline-end: var(--text-field-suffix-padding-right, 0px);
+      }
+      .mdc-text-field__affix--suffix,
+      .mdc-text-field--with-leading-icon {
+        padding-inline-start: var(--text-field-suffix-padding-left, 0px);
+        padding-inline-end: var(--text-field-suffix-padding-right, 12px);
```

Comment:
0 and 12 are now flipped?

Reply:
Sorry. Made a mess there. Separated, fixed, verified.